driver-core/src/main/java/com/datastax/driver/core/Responses.java

```diff
@@ -380,6 +384,17 @@ static Metadata decode(ByteBuf body, boolean withPkIndices, ProtocolVersion prot
                     EnumSet<Flag> flags = Flag.deserialize(body.readInt());
                     int columnCount = body.readInt();
 
+                    ByteBuffer state = null;
+                    if (flags.contains(Flag.HAS_MORE_PAGES))
+                        state = CBUtil.readValue(body);
+
+                    MD5Digest resultMetadataId = null;
+                    if (flags.contains(Flag.METADATA_CHANGED)) {
+                        assert protocolVersion == ProtocolVersion.V5 : "METADATA_CHANGED flag is supported starting from v5";
```

Comment:
Is it possible that we would ever receive from Cassandra the `METADATA_CHANGED` with protocol < V5? Maybe that check isn't necessary?